### PR TITLE
ZIP export, second try

### PR DESCRIPTION
**Pull request by LemonAndroid**
Before submitting pull request, please ensure that:

- [ ] Every commit has message which describes it
- [ ] Every commit is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [ ] Any code changes come with testcase
- [ ] Any new functionality is covered by user documentation

### CODE REVIEW

**Comment by LemonAndroid** [comment]
@nijel It is incomplete, it will take quite some time to complete it.

**Comment by nijel** [comment]
This pull request **introduces 1 alert** when merging 56b92e45be0da04837e2dd5d3aaa64ba969480ef into 6cfa5a6c200d150336b01d5fd9cfd680d9874096 - [view on LGTM.com](https://lgtm.com/projects/g/WeblateOrg/weblate/rev/pr-ff12a3c4480d431c46d7f1efeb6eea6f3e325196)

**new alerts:**

* 1 for Syntax error

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by nijel** [comment]
This pull request **introduces 3 alerts** when merging c4dbad1ecf02655b8cdb42790ea7fa01b49b5292 into 07a1caa8526e85babd18adf5cdb10262f48d8938 - [view on LGTM.com](https://lgtm.com/projects/g/WeblateOrg/weblate/rev/pr-977a67117b8c04b3282973f949e206594daa4d97)

**new alerts:**

* 1 for Testing equality to None
* 1 for Unused import
* 1 for Syntax error

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by nijel** [comment]
This pull request **introduces 3 alerts** when merging 0686cb33211d19546003ff565641ea96bab6dbed into 373293292d830a48acde2ff7a96f841c85a55447 - [view on LGTM.com](https://lgtm.com/projects/g/WeblateOrg/weblate/rev/pr-7de7fa2b951ad79981d48b2e62a8b24442c2347a)

**new alerts:**

* 1 for Testing equality to None
* 1 for Unused import
* 1 for Syntax error

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: weblate/trans/templatetags/downloads.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    closing_tag = "</a>"
+
+    return opening_tag + link_text(context) + closing_tag
+
+
+def link_text(context):
+    exporter = context.get('exporter')
+
+    return _("Original" if exporter == None  else exporter['name'])
```
Comment (comradekingu) [inline weblate/trans/templatetags/downloads.py]:
```suggestion
    return _("Original" if exporter == None else exporter['name'])
```

File: weblate/trans/templatetags/downloads.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+from django import template
+from django.utils.translation import gettext as _
+from django.urls import reverse
+
+register = template.Library()
+
+@register.simple_tag(takes_context=True)
+def translation_download_url(context):
+    #import pdb; pdb.set_trace()
```
Comment (comradekingu) [inline weblate/trans/templatetags/downloads.py]:
```suggestion
    # import pdb; pdb.set_trace()
```

File: weblate/trans/templatetags/test_downloads.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+from weblate.trans.templatetags.downloads import link_text, translation_download_link, translation_download_url
+
+class Struct:
+    def __init__(self, **entries):
+        self.__dict__.update(entries)
+
+def struct(dict):
+    return Struct(**dict)
+
+class RenderContextMock():
```
Comment (comradekingu) [inline weblate/trans/templatetags/test_downloads.py]:
```suggestion

class RenderContextMock():
```

File: weblate/trans/templatetags/downloads.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
+from django import template
+from django.utils.translation import gettext as _
+from django.urls import reverse
+
+register = template.Library()
+
+@register.simple_tag(takes_context=True)
```
Comment (comradekingu) [inline weblate/trans/templatetags/downloads.py]:
```suggestion

@register.simple_tag(takes_context=True)
```

File: weblate/trans/templatetags/downloads.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    project = context.get('project')
+
+    kwargs = {}
+    kwargs['project'] = project.slug
+    component = context.get('component')
+    if component != None:
+        kwargs['component'] = component.slug
+
+    url = reverse('download_translation', kwargs=kwargs)
```
Comment (comradekingu) [inline weblate/trans/templatetags/downloads.py]:
```suggestion
    url = reverse('download_translation', kwargs=kwargs)
```

File: weblate/trans/templatetags/downloads.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    if language != None:
+        query_params.append(f'lang={language.code}')
+
+    if len(query_params) > 0:
+        url += '?'
+        url += "&".join(query_params)
+
+    return url
+
+@register.simple_tag(takes_context=True)
```
Comment (comradekingu) [inline weblate/trans/templatetags/downloads.py]:
```suggestion

@register.simple_tag(takes_context=True)
```

File: weblate/trans/templatetags/downloads.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+def translation_download_link_todo(context, todo=False):
+    translation_download_link(context, True)
+
+@register.simple_tag(takes_context=True)
```
Comment (comradekingu) [inline weblate/trans/templatetags/downloads.py]:
```suggestion

@register.simple_tag(takes_context=True)
```

File: weblate/trans/templatetags/test_downloads.py (hunk 3 above)
Comment (comradekingu) [inline weblate/trans/templatetags/test_downloads.py]:
```suggestion

class RenderContextMock():
```

**Comment by nijel** [comment]
This pull request **introduces 5 alerts** when merging 4253eeb391225c8f19fa95a2c154e7dc3c7f98a2 into ec4514f25a5493bfb53d56f1cb51c4760fba9aba - [view on LGTM.com](https://lgtm.com/projects/g/WeblateOrg/weblate/rev/pr-27c7915f34f6edcbb3e41553f9aef1ab97eb5dea)

**new alerts:**

* 2 for Testing equality to None
* 2 for Wrong number of arguments in a call
* 1 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by comradekingu** [comment]
@LemonAndroid Looked like this only needs fixing the LGTM issues, can you fix those?

**Comment by LemonAndroid** [comment]
@comradekingu i'm unsure, if someone else wants to finish and collect the bounty or put it back into the pool, i'm fine. It's not like i'm planning to have a new try at this soon.

**Pull request closed without merging**